client/analytics/settings/setting.js

```diff
@@ -0,0 +1,141 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import { uniqueId } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import './setting.scss';
+import sanitizeHTML from 'lib/sanitize-html';
+
+class Setting extends Component {
+	renderInput = () => {
+		const { handleChange, name, inputType, options, value } = this.props;
+		const id = uniqueId( name );
+
+		switch ( inputType ) {
+			case 'checkboxGroup':
+				return options.map(
+					optionGroup =>
+						optionGroup.options.length > 0 && (
+							<div className="woocommerce-setting__options-group" key={ optionGroup.key }>
+								{ optionGroup.label && (
+									<span className="woocommerce-setting__options-group-label">
+										{ optionGroup.label }
+									</span>
+								) }
+								{ this.renderCheckboxOptions( optionGroup.options ) }
+							</div>
+						)
+				);
+			case 'checkbox':
+				return this.renderCheckboxOptions( options );
+			case 'text':
+			default:
+				return (
+					<input id={ id } type="text" name={ name } onChange={ handleChange } value={ value } />
+				);
+		}
+	};
+
+	renderCheckboxOptions( options ) {
+		const { handleChange, name, value } = this.props;
+
+		return options.map( option => {
+			const id = uniqueId( name + '-' + option.value );
+			return (
+				<label htmlFor={ id } key={ option.value }>
+					<input
+						id={ id }
+						type="checkbox"
+						name={ name }
+						onChange={ handleChange }
+						aria-label={ option.description }
+						checked={ value && value.includes( option.value ) }
+						value={ option.value }
+					/>
+					{ option.label }
+				</label>
+			);
+		} );
+	}
+
+	render() {
+		const { helpText, label } = this.props;
+
+		return (
+			<div className="woocommerce-setting">
+				<div className="woocommerce-setting__label">{ label }</div>
+				<div className="woocommerce-setting__options">
+					{ this.renderInput() }
+					{ helpText && (
+						<span
+							className="woocommerce-setting__help"
+							dangerouslySetInnerHTML={ sanitizeHTML( helpText ) }
```

Comment:
If the help text for the analytics settings used `interpolateComponents()` instead of markup, would we need `dangerouslySetInnerHTML` and `sanitizeHTML`?

Reply:
Updated to use `interpolateComponents()`.